lightning/src/util/events.rs

```diff
@@ -67,6 +68,38 @@ pub enum PaymentPurpose {
 	SpontaneousPayment(PaymentPreimage),
 }
 
+#[derive(Clone, Debug)]
+/// Some information provided on the closure source of the channel halting.
+pub enum ClosureReason {
+	/// Closure generated from receiving a peer error message by ChannelManager::handle_error
+	CounterpartyForceClosed {
+		/// The error is coming from the peer, there *might* be a human-readable msg
```

Comment:
Grammar nit: "The error which the peer sent us". This needs the same warning as `ErrorMessage::data`:
```
	/// The string should be sanitized before it is used (e.g. emitted to logs
	/// or printed to stdout).  Otherwise, a well crafted error message may trigger a security
	/// vulnerability in the terminal emulator or the logging subsystem.
```